server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/index/StarTreeValues.java

```diff
@@ -0,0 +1,173 @@
+            starTreeMetadata.getMetrics(),
+            new StarTreeFieldConfiguration(
+                starTreeMetadata.getMaxLeafDocs(),
+                skipStarNodeCreationInDims,
+                starTreeMetadata.getStarTreeBuildMode()
+            )
+        );
+
+        this.root = StarTreeFactory.createStarTree(compositeIndexIn, starTreeMetadata);
+
+        // get doc id set iterators for metrics and dimensions
+        dimensionDocValuesIteratorMap = new LinkedHashMap<>();
+        metricDocValuesIteratorMap = new LinkedHashMap<>();
+
+        // get doc id set iterators for dimensions
+        for (String dimension : starTreeMetadata.getDimensionFields()) {
+            SortedNumericDocValues dimensionSortedNumericDocValues = null;
+            if (readState != null) {
+                FieldInfo dimensionfieldInfo = readState.fieldInfos.fieldInfo(
+                    fullyQualifiedFieldNameForStarTreeDimensionsDocValues(starTreeField.getName(), dimension)
+                );
+                if (dimensionfieldInfo != null) {
+                    dimensionSortedNumericDocValues = compositeDocValuesProducer.getSortedNumeric(dimensionfieldInfo);
+                }
+            }
+
+            dimensionDocValuesIteratorMap.put(dimension, getSortedNumericDocValues(dimensionSortedNumericDocValues));
+        }
+
+        // get doc id set iterators for metrics
+
+        for (Metric metric : starTreeMetadata.getMetrics()) {
+            for (MetricStat metricStat : metric.getMetrics()) {
+                String metricFullName = fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+                    starTreeField.getName(),
+                    metric.getField(),
+                    metricStat.getTypeName()
+                );
+
+                SortedNumericDocValues metricSortedNumericDocValues = null;
+                if (readState != null) {
+                    FieldInfo metricFieldInfo = readState.fieldInfos.fieldInfo(metricFullName);
+                    if (metricFieldInfo != null) {
+                        metricSortedNumericDocValues = compositeDocValuesProducer.getSortedNumeric(metricFieldInfo);
+                    }
+                }
+                metricDocValuesIteratorMap.put(metricFullName, getSortedNumericDocValues(metricSortedNumericDocValues));
+            }
+        }
+
+        // create star-tree attributes map
+        attributes = new HashMap<>();
+        attributes.put(SEGMENT_DOCS_COUNT, String.valueOf(starTreeMetadata.getSegmentAggregatedDocCount()));
+        attributes.put(STAR_TREE_DOCS_COUNT, String.valueOf(starTreeMetadata.getStarTreeDocCount()));
+
+    }
+
+    @Override
+    public CompositeIndexValues getValues() {
+        return this;
+    }
+
+    public StarTreeField getStarTreeField() {
+        return starTreeField;
+    }
+
+    public StarTreeNode getRoot() {
+        return root;
+    }
+
+    public Map<String, DocIdSetIterator> getDimensionDocValuesIteratorMap() {
```

Comment:
Can this take the dimension name in the input only? Do we need to return the map?